### ISSUE TITLE
Collections larger than 10g will fail to load

### ISSUE THREAD

**Issue by RipperTs** [qdrant/qdrant#2324]
I have a collection: metallurgy, when its size exceeds 10gb, I reopen QdRant
It will never start successfully!

Note, I am not a docker installation, but a directly run binary

I am windows platform

**Comment by RipperTs** [comment]
It may be due to the relatively large amount of data, and the startup is successful after 20 minutes

**Comment by timvisee** [comment]
Are you saying that when you restart your Qdrant instance, that has a huge collection, that it takes about 20 minutes for it to fully start?

Also, does it not handle any requests before these 20 minutes?

Note that Docker on Windows is run in a VM. That can have quite a significant effect on performance. Especially if the resources assigned to the VM are limited.

**Comment by RipperTs** [comment]
yes below is my startup log
```
[2023-07-25T08:49:58.682Z WARN  qdrant::settings] Config file not found: config/config
[2023-07-25T08:49:58.682Z WARN  qdrant::settings] Config file not found: config/development
[2023-07-25T08:49:58.683Z INFO  storage::content_manager::consensus::persistent] Loading raft state from ./storage\raft_state
[2023-07-25T08:49:58.687Z INFO  storage::content_manager::toc] Loading collection: meetings-m3e
Recovering collection meetings-m3e [00:00:00] █████████████████████████████████████Recovering collection meetings-m3e [00:00:00] █████████████████████████████████████████████████████████████ 1/1 (eta:0s)[2023-07-25T08:50:00.228Z INFO  storage::content_manager::toc] Loading collection: metallurgy
Recovering collection metallurgy [00:00:00] ██████████████████████████████████████Recovering collection metallurgy [00:00:00] ███████████████████████████████████████████████████████████████ 1/1 (eta:0s)[2023-07-25T09:14:12.716Z INFO  storage::content_manager::toc] Loading collection: my_documents
Recovering collection my_documents [00:00:00] █████████████████████████████████████Recovering collection my_documents [00:00:00] █████████████████████████████████████████████████████████████ 1/1 (eta:0s)[2023-07-25T09:14:12.926Z INFO  qdrant] Distributed mode disabled
[2023-07-25T09:14:12.926Z INFO  qdrant] Telemetry reporting enabled, id: d2a8a8b1-93d6-4f53-a63c-239bd8a03d54
[2023-07-25T09:14:12.927Z INFO  qdrant::tonic] Qdrant gRPC listening on 6334
[2023-07-25T09:14:12.927Z INFO  qdrant::tonic] TLS disabled for gRPC API
[2023-07-25T09:14:12.927Z WARN  qdrant::actix] Static content folder for Web UI './static' does not exist
[2023-07-25T09:14:12.928Z INFO  qdrant::actix] TLS disabled for REST API
[2023-07-25T09:14:12.929Z INFO  qdrant::actix] Qdrant HTTP listening on 6333
[2023-07-25T09:14:12.929Z INFO  actix_server::builder] Starting 31 workers
```
I didn't use docker, directly download qdrant.exe to run

**Comment by RipperTs** [comment]
> Are you saying that when you restart your Qdrant instance, that has a huge collection, that it takes about 20 minutes for it to fully start?您是说当您重新启动 Qdrant 实例时，它有一个巨大的集合，需要大约 20 分钟才能完全启动？
> 
> Also, does it not handle any requests before these 20 minutes?另外，在这 20 分钟之前它不会处理任何请求吗？
> 
> Note that Docker on Windows is run in a VM. That can have quite a significant effect on performance. Especially if the resources assigned to the VM are limited.请注意，Windows 上的 Docker 是在虚拟机中运行的。这会对性能产生相当大的影响。特别是当分配给虚拟机的资源有限时。

yes below is my startup log

```
[2023-07-25T08:49:58.682Z WARN  qdrant::settings] Config file not found: config/config
[2023-07-25T08:49:58.682Z WARN  qdrant::settings] Config file not found: config/development
[2023-07-25T08:49:58.683Z INFO  storage::content_manager::consensus::persistent] Loading raft state from ./storage\raft_state
[2023-07-25T08:49:58.687Z INFO  storage::content_manager::toc] Loading collection: meetings-m3e
Recovering collection meetings-m3e [00:00:00] █████████████████████████████████████Recovering collection meetings-m3e [00:00:00] █████████████████████████████████████████████████████████████ 1/1 (eta:0s)[2023-07-25T08:50:00.228Z INFO  storage::content_manager::toc] Loading collection: metallurgy
Recovering collection metallurgy [00:00:00] ██████████████████████████████████████Recovering collection metallurgy [00:00:00] ███████████████████████████████████████████████████████████████ 1/1 (eta:0s)[2023-07-25T09:14:12.716Z INFO  storage::content_manager::toc] Loading collection: my_documents
Recovering collection my_documents [00:00:00] █████████████████████████████████████Recovering collection my_documents [00:00:00] █████████████████████████████████████████████████████████████ 1/1 (eta:0s)[2023-07-25T09:14:12.926Z INFO  qdrant] Distributed mode disabled
[2023-07-25T09:14:12.926Z INFO  qdrant] Telemetry reporting enabled, id: d2a8a8b1-93d6-4f53-a63c-239bd8a03d54
[2023-07-25T09:14:12.927Z INFO  qdrant::tonic] Qdrant gRPC listening on 6334
[2023-07-25T09:14:12.927Z INFO  qdrant::tonic] TLS disabled for gRPC API
[2023-07-25T09:14:12.927Z WARN  qdrant::actix] Static content folder for Web UI './static' does not exist
[2023-07-25T09:14:12.928Z INFO  qdrant::actix] TLS disabled for REST API
[2023-07-25T09:14:12.929Z INFO  qdrant::actix] Qdrant HTTP listening on 6333
[2023-07-25T09:14:12.929Z INFO  actix_server::builder] Starting 31 workers
```

I didn't use docker, directly download qdrant.exe to run

**Comment by timvisee** [comment]
Based on 

```
[2023-07-25T08:50:00.228Z INFO  storage::content_manager::toc] Loading collection: metallurgy
Recovering collection metallurgy [00:00:00] ██████████████████████████████████████ 1/1 (eta:0s)
[2023-07-25T09:14:12.716Z INFO  storage::content_manager::toc] Loading collection: my_documents
```

I see that it indeed takes 25 minutes to load/recover `my_documents`. That is indeed extremely long.

Could you share the status of the collection after startup? You can find that by visiting: `http://localhost:6333/collections/my_documents`

Also, what is the size of the whole collection on disk?

**Comment by RipperTs** [comment]
> Based on 基于
> 
> ```
> [2023-07-25T08:50:00.228Z INFO  storage::content_manager::toc] Loading collection: metallurgy
> Recovering collection metallurgy [00:00:00] ██████████████████████████████████████ 1/1 (eta:0s)
> [2023-07-25T09:14:12.716Z INFO  storage::content_manager::toc] Loading collection: my_documents
> ```
> 
> I see that it indeed takes 25 minutes to load/recover `my_documents`. That is indeed extremely long.我发现加载/恢复 `my_documents` 确实需要 25 分钟。这确实是极其漫长的。
> 
> Could you share the status of the collection after startup? You can find that by visiting: `http://localhost:6333/collections/my_documents`能否分享一下启动后集合的状态？您可以通过访问： `http://localhost:6333/collections/my_documents` 找到它
> 
> Also, what is the size of the whole collection on disk?另外，磁盘上整个集合的大小是多少？

No, it's actually loading the `metallurgy` collection that takes a long time, not `my_documents `

Here is the state of the `metallurgy` collection:
```json
{
  "result": {
    "status": "green",
    "optimizer_status": "ok",
    "vectors_count": 2694891,
    "indexed_vectors_count": 1197601,
    "points_count": 2694891,
    "segments_count": 9,
    "config": {
      "params": {
        "vectors": {
          "size": 768,
          "distance": "Cosine"
        },
        "shard_number": 1,
        "replication_factor": 1,
        "write_consistency_factor": 1,
        "on_disk_payload": true
      },
      "hnsw_config": {
        "m": 16,
        "ef_construct": 100,
        "full_scan_threshold": 10000,
        "max_indexing_threads": 0,
        "on_disk": false
      },
      "optimizer_config": {
        "deleted_threshold": 0.2,
        "vacuum_min_vector_number": 1000,
        "default_segment_number": 0,
        "max_segment_size": null,
        "memmap_threshold": null,
        "indexing_threshold": 20000,
        "flush_interval_sec": 5,
        "max_optimization_threads": 1
      },
      "wal_config": {
        "wal_capacity_mb": 32,
        "wal_segments_ahead": 0
      },
      "quantization_config": null
    },
    "payload_schema": {
      
    }
  },
  "status": "ok",
  "time": 0.0000281
}
```

The current size of the entire collection on disk is `10.5GB`  
Also, I made a snapshot of the metallurgy collection, which is `9.28GB` in size

**Comment by timvisee** [comment]
> No, it's actually loading the `metallurgy` collection that takes a long time, not `my_documents `

You're totally right! My bad.

Thank you for sharing the collection status. Based on that your vector data and index is always loaded in memory. It must be fully loaded before it can accept any requests.

Based on this, I have a few questions:

1. 10GB is relatively large. Do you have enough RAM available to fit the whole collection?
2. Is it possible that it is heavily swapping?
3. Is your disk fast enough? A very slow or network attached disk can have a significant effect on this.

**Comment by RipperTs** [comment]
> > No, it's actually loading the `metallurgy` collection that takes a long time, not `my_documents `不，它实际上是加载需要很长时间的 `metallurgy` 集合，而不是 `my_documents `
> 
> You're totally right! My bad.你完全正确！我的错。
> 
> Thank you for sharing the collection status. Based on that your vector data and index is always loaded in memory. It must be fully loaded before it can accept any requests.感谢您分享收集状态。基于此，您的矢量数据和索引始终加载到内存中。它必须完全加载才能接受任何请求。
> 
> Based on this, I have a few questions:基于此，我有几个问题：
> 
> 1. 10GB is relatively large. Do you have enough RAM available to fit the whole collection?10GB算是比较大的了。您是否有足够的 RAM 来容纳整个集合？
> 2. Is it possible that it is heavily swapping?是否有可能大量交换？
> 3. Is your disk fast enough? A very slow or network attached disk can have a significant effect on this.你的磁盘足够快吗？速度非常慢的磁盘或网络连接的磁盘可能会对此产生重大影响。

My current server memory is `48GB`, and the memory consumed to load the entire collection is around `16GB`, so it's perfectly fine.

Also, it's all local ssd disks, so it should be ok speed wise.

The phenomenon of interest is that the program loads the entire collection into memory, a process that takes about just about 5 minutes or so, and the way I can tell is that the `qdrant` process takes up `16GB` of memory and then stops adding more, and for the rest of the time it's not clear what the `qdrant` program is doing.

Another question I'd like to ask is whether or not running the `qdrant.exe` binary directly on `Windows` is unstable relative to `docker` startup? Because it occasionally forces the process to exit causing me to restart `qdrant` to fix it, and restarting consumes a lot of my time.
Do you have any good suggestions for my current situation?

**Comment by RipperTs** [comment]
I upgraded to version 1.4.0 by replacing the qdrant.exe file, but I found that I can no longer start it, it keeps telling me that the collection is loading, and it takes more than an hour to load.

**Comment by timvisee** [comment]
> I upgraded to version 1.4.0 by replacing the qdrant.exe file, but I found that I can no longer start it, it keeps telling me that the collection is loading, and it takes more than an hour to load.

Do you see a progress bar like this again, which takes a long time to reach 100%?

[2023-07-25T08:50:00.228Z INFO  storage::content_manager::toc] Loading collection: metallurgy
Recovering collection metallurgy [00:00:00] ████████████████████████████                                 0/1 (eta:0s)

**Comment by RipperTs** [comment]
> 

```
[2023-07-25T08:50:00.228Z INFO  storage::content_manager::toc] Loading collection: metallurgy
Recovering collection metallurgy
```
What I saw was this, no results for almost an hour, then I canceled manually, and the replacement would be the last 1 3 version which took 25 minutes to run successfully.

I speculate that the new version may have been affected by the optimizations made for memory management; I'm using version 1.4.0 of the program, and the memory is around 8.8GB after startup.
With the 1.3 version of the program, the memory footprint is about 16GB after startup.

I can provide a converted vector data snapshot package for testing if needed.
